common/src/main/java/com/vexsoftware/votifier/util/standalone/StandaloneVotifierPlugin.java

```diff
@@ -1,65 +1,47 @@
-import com.vexsoftware.votifier.net.protocol.VotifierProtocolDifferentiator;
-import io.netty.bootstrap.ServerBootstrap;
+import com.vexsoftware.votifier.platform.scheduler.ScheduledExecutorServiceVotifierScheduler;
+import com.vexsoftware.votifier.platform.scheduler.VotifierScheduler;
 import io.netty.channel.Channel;
-import io.netty.channel.ChannelFuture;
-import io.netty.channel.ChannelInitializer;
-import io.netty.channel.nio.NioEventLoopGroup;
-import io.netty.channel.socket.nio.NioServerSocketChannel;
-import io.netty.channel.socket.nio.NioSocketChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import java.net.InetSocketAddress;
 import java.security.Key;
 import java.security.KeyPair;
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.function.Consumer;
 
-public class StandaloneVotifierPlugin implements VotifierPlugin, VoteHandler {
+public class StandaloneVotifierPlugin implements VotifierPlugin {
     private final Map<String, Key> tokens;
     private final VoteReceiver receiver;
     private final KeyPair v1Key;
     private final InetSocketAddress bind;
-    private NioEventLoopGroup group;
+    private final VotifierScheduler scheduler;
+    private VotifierServerBootstrap bootstrap;
 
     public StandaloneVotifierPlugin(Map<String, Key> tokens, VoteReceiver receiver, KeyPair v1Key, InetSocketAddress bind) {
         this.receiver = receiver;
         this.bind = bind;
         this.tokens = Collections.unmodifiableMap(new HashMap<>(tokens));
         this.v1Key = v1Key;
+        this.scheduler = new ScheduledExecutorServiceVotifierScheduler(Executors.newScheduledThreadPool(1));
     }
 
-    public Throwable start() {
-        group = new NioEventLoopGroup(2);
-
-        ChannelFuture future = new ServerBootstrap()
-                .channel(NioServerSocketChannel.class)
-                .group(group)
-                .childHandler(new ChannelInitializer<NioSocketChannel>() {
-                    @Override
-                    protected void initChannel(NioSocketChannel channel) {
-                        channel.attr(VotifierSession.KEY).set(new VotifierSession());
-                        channel.attr(VotifierPlugin.KEY).set(StandaloneVotifierPlugin.this);
-                        channel.pipeline().addLast("greetingHandler", new VotifierGreetingHandler());
-                        channel.pipeline().addLast("protocolDifferentiator", new VotifierProtocolDifferentiator(false, v1Key != null));
-                        channel.pipeline().addLast("voteHandler", new VoteInboundHandler(StandaloneVotifierPlugin.this));
-                    }
-                })
-                .bind(bind)
-                .syncUninterruptibly();
-
-        if (!future.isSuccess()) {
-            return future.cause();
-        } else {
-            return null;
+    public void start(Consumer<Throwable> error) {
```

Comment:
Now _technically_ we should maintain backwards compat. and have a 0 argument start

Reply:
Should also handle the case where error is null